### PR TITLE
client api

### PR DESCRIPTION
**Pull request by nathan-downs**
## Test plan

sg start
verify in bigquery

### CODE REVIEW

File: cmd/frontend/graphqlbackend/telemetry.graphql [hunk 1]
```diff
@@ -90,11 +90,11 @@ input TelemetryEventParametersInput {
...
     By default, this metadata is assumed to be unsafe for export from an instance.
     """
-    privateMetadata: JSONValue
+    privateMetadata: JSONValue!
     """
     Billing-related metadata.
     """
-    billingMetadata: TelemetryEventBillingMetadataInput
+    billingMetadata: TelemetryEventBillingMetadataInput!
 }
```
Comment (bobheadxi) [inline cmd/frontend/graphqlbackend/telemetry.graphql:98]: is making these required intentional? they were left optional intentionally, as not all events need `privateMetadata` or billing-related attributes (maybe?)

Comment (nathan-downs) [reply cmd/frontend/graphqlbackend/telemetry.graphql:98]: That was an oversight, I was flipping those back and forth when I was testing. I'll walk that back, good catch!

File: client/shared/src/platform/context.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ import type { ExecutableExtension } from '../api/extension/activation'
 import type { Scalars } from '../graphql-operations'
 import type { Settings, SettingsCascadeOrError } from '../settings/settings'
 import type { TelemetryService } from '../telemetry/telemetryService'
+import type { TelemetryServiceV2 } from '../telemetry/telemetryServiceV2'
```
Comment (bobheadxi) [inline client/shared/src/platform/context.ts:14]: The contract should just be https://github.com/sourcegraph/telemetry I think

File: client/browser/src/shared/tracking/telemetryRecorder.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    }
+
+    /**
+     * Implements {@link TelemetryService}.
+     *
+     * @todo Handle arbitrary action IDs.
+     *
+     * @param eventName The ID of the action executed.
+     */
+    public async record(eventName: string, eventProperties?: any, publicArgument?: any): Promise<void> {
```
Comment (bobheadxi) [inline client/browser/src/shared/tracking/telemetryRecorder.tsx:138]: We shouldn't be exposing `eventProperties?: any, publicArgument?: any` anywhere, we should only be exposing https://github.com/sourcegraph/telemetry interfaces that dictate the shape of the SDK that developers interact with when making new events

File: client/web/src/user/settings/backend.tsx [hunk 4]
```diff
@@ -203,3 +207,89 @@ function createEvent(event: string, eventProperties?: unknown, publicArgument?:
         hashedLicenseKey: window.context?.hashedLicenseKey,
     }
 }
+
+// Log events in batches
+const batchedTelemetryEvents = new Subject<TelemetryEventInput>()
```
Comment (bobheadxi) [inline client/web/src/user/settings/backend.tsx:212]: @nathan-downs the plan is to do this in sourcegraph/telemetry, right? Let's keep this implementation bare-bones for now

**Comment by nathan-downs** [comment]
Closing in favor of this pr: https://github.com/sourcegraph/sourcegraph/pull/57680

**Pull request closed without merging**